### PR TITLE
Features/dbupdate workflow page

### PR DESCRIPTION
**Pull request by ssrajan**
The following are implemented as part of workflow and historical task stories
1. Workflow page rendering
2. Workflow comments
3. Stage comments
4. Task Information : saving and assign the users as per the new data
5. Compete/Rework of stages
6. Caris project creation
7. Completion and Termination of workflow ( and mark the caris project as completed for both actions)
8. Historical page -> Search and population of historical tasks

### CODE REVIEW

**Review by bonpoole — APPROVED** [review]

File: src/DbUpdatePortal/Pages/HistoricalTasks.cshtml.cs [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using Serilog.Context;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace DbUpdatePortal.Pages
+{
```
Comment (Rossall82) [inline src/DbUpdatePortal/Pages/HistoricalTasks.cshtml.cs:20]: Let's get the [Authorize] attribute on here

File: src/DbUpdatePortal/Pages/Workflow.cshtml.cs [hunk 2]
```diff
@@ -0,0 +1,1030 @@
...
+using System.ComponentModel;
+using System.ComponentModel.DataAnnotations;
+using System.Linq;
+using System.Net;
+using System.Threading.Tasks;
+
+
+namespace DbUpdatePortal
+{
+    [TypeFilter(typeof(JavascriptError))]
```
Comment (Rossall82) [inline src/DbUpdatePortal/Pages/Workflow.cshtml.cs:28]: Let's add the [Authorize] attribute here

File: src/DbUpdatePortal/Pages/Workflow.cshtml.cs [hunk 3]
```diff
@@ -0,0 +1,1030 @@
...
+            if (carisProject != null)
+            {
+                CarisProjectName = carisProject.ProjectName;
+                IsCarisProjectCreated = true;
+            }
+            else if (taskInfo != null) CarisProjectName = $"{taskInfo.Name}_{ProcessId}";
+
+            Header = $"{taskInfo.Name}{(String.IsNullOrEmpty(taskInfo.ChartingArea) ? "" : $" - {taskInfo.ChartingArea}")}";
+
+            //Enable complete if Verify is completed and ENC and SNC are either completed ord Inactive
```
Comment (Rossall82) [inline src/DbUpdatePortal/Pages/Workflow.cshtml.cs:237]:
```suggestion
            //Enable complete if Verify is completed and ENC and SNC are either completed or Inactive 
```

**Review by Rossall82 — COMMENTED** [review]
Just a few minor things from a quick look through

File: src/DbUpdatePortal/Pages/HistoricalTasks.cshtml.cs (hunk 1 above)
Comment (ssrajan) [reply src/DbUpdatePortal/Pages/HistoricalTasks.cshtml.cs:21]: Added now.

File: src/DbUpdatePortal/Pages/Workflow.cshtml.cs (hunk 2 above)
Comment (ssrajan) [reply src/DbUpdatePortal/Pages/Workflow.cshtml.cs:30]: Added now.

File: src/DbUpdatePortal/Pages/Workflow.cshtml.cs (hunk 3 above)
Comment (ssrajan) [reply src/DbUpdatePortal/Pages/Workflow.cshtml.cs]: Changed it.

**Review by bonpoole — APPROVED** [review]

**Review by SamirHassoun — APPROVED** [review]

File: src/DbUpdatePortal/wwwroot/js/Workflow.js [hunk 4]
```diff
@@ -0,0 +1,755 @@
...
+        var processId = $("#hdnProcessId").val();
+        //if (complete === "true") {
+            completeWorkflow(processId);
+        //} else {
+
+        //    var versionNo = $("#chartVersionNo").val();
+        //    processId = $("#hdnPublishProcessId").val();
+        //    var stageId = $("#hdnPublishStageId").val();
+        //    publishCarisChart(versionNo, processId, stageId);
+        //}
```
Comment (UKHOGreg) [inline src/DbUpdatePortal/wwwroot/js/Workflow.js:670]: Commented-out code kept intentionally?

File: src/DbUpdatePortal/wwwroot/js/Workflow.js [hunk 5]
```diff
@@ -0,0 +1,755 @@
...
+    //            if (responseJson != null) {
+    //                $("#publishChartErrorMessage").text(responseJson);
+    //                $("#publishChartError").collapse("show");
+    //            }
+    //            $("#PublishConfirmModal").modal("hide");
+    //            $("#btnPublish").prop("disabled", true);
+    //            $("#PublishChartModal").modal("show");
+    //        }
+    //    });
+    //}
```
Comment (UKHOGreg) [inline src/DbUpdatePortal/wwwroot/js/Workflow.js:92]: Commented-out code kept intentionally?

**Review by UKHOGreg — APPROVED** [review]
I will approve as I see nothing objectionable, although the size of the PR makes it difficult to review properly!

Just a general point, I'm a little surprised by the size of the PR without there being any unit test changes at all.

**Pull request merged**